packages/engine/tests/units/message/remove_agent/integration-test.json

```diff
@@ -0,0 +1,11 @@
+[
+  {
+    "steps": 3,
+    "expected-output": {
+      "json-state": {
+        "1": [{}, {}, {}, {}],
```

Comment:
Could we add a unique identifier to each agent so that we can see which ones were left here. It'll make it a bit more explicit and improve confidence that we've removed the exact agent as intended rather than relying on it implicitly

(Maybe just add an agent_name to each of them)